x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -830,6 +833,41 @@ static void validateRealmSettings(Settings settings) {
         }
     }
 
+    static void validateForFips(Settings settings, XPackLicenseState licenseState) {
+        final List<String> validationErrors = new ArrayList<>();
+        Settings keystoreTypeSettings = settings.filter(k -> k.endsWith("keystore.type"))
+            .filter(k -> settings.get(k).equalsIgnoreCase("jks"));
+        if (keystoreTypeSettings.isEmpty() == false) {
+            validationErrors.add("JKS Keystores cannot be used in a FIPS 140 compliant JVM. Please " +
+                "revisit [" + keystoreTypeSettings.toDelimitedString(',') + "] settings");
+        }
+        Settings keystorePathSettings = settings.filter(k -> k.endsWith("keystore.path"))
+            .filter(k -> settings.hasValue(k.replace(".path", ".type")) == false);
+        // Default Keystore type is JKS in JDK8 and PKCS12 in > JDK9 if not explicitly set
```

Comment:
This is not the case. We explicitly set the keystore type to JKS, and don't depend on the JVM default.
https://github.com/elastic/elasticsearch/blob/9945d5cf35183263184e518aa73a72959fb15a0e/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationSettings.java#L245-L252

Perhaps we should call `inferKeyStoreType` here instead.